packages/next/src/lib/metadata/resolvers/resolve-url.ts

```diff
@@ -93,10 +93,33 @@ function resolveAbsoluteUrlWithPathname(
     canonicalUrl = result.pathname === '/' ? result.origin : result.href
   }
 
-  // Add trailing slash if it's enabled
-  return trailingSlash && !canonicalUrl.endsWith('/')
-    ? `${canonicalUrl}/`
-    : canonicalUrl
+  let isRelative = canonicalUrl.startsWith('/')
```

Comment:
is this an exhaustive relative check? I think `./foo` is technically a valid way to specify a relative URL

Reply:
It's already always starting with `/`, I'll push some comment updates to explain here